main.js

```diff
@@ -77,20 +77,20 @@ const log = report => {
 	process.exit(report.errorCount === 0 ? 0 : 1);
 };
 
+const files = (report, predicate) => report.results
+	.filter(predicate)
```

Comment:
0: seems fair to me, better safe than sorry
1-2: I had the assumption that the warnings should only open if there er no errors. That's why I didn't implement the sorting, and solved it with a `if` case.